### PR TITLE
added fsl and spm use flags

### PR DESCRIPTION
**Pull request by TheChymera**
Scond try :)

### CODE REVIEW

File: sci-libs/nipype/nipype-9999.ebuild [hunk 1]
```diff
@@ -18,12 +18,14 @@ EGIT_REPO_URI="https://github.com/nipy/nipype"
...
-IUSE=""
+IUSE="fsl spm"
 
 DEPEND="
 	dev-python/setuptools[${PYTHON_USEDEP}]
 	dev-python/numpy[${PYTHON_USEDEP}]
-	>=sci-libs/nibabel-1.2[${PYTHON_USEDEP}]"
+	>=sci-libs/nibabel-1.2[${PYTHON_USEDEP}]
+	fsl? ( sci-biology/fsl )
+	spm ( sc-biology/spm )"
```
Comment (junghans) [inline sci-libs/nipype/nipype-9999.ebuild]: Question mark mssing here!

Comment (junghans) [inline sci-libs/nipype/nipype-9999.ebuild]: And typo: `sc-biology` -> `sci-biology`

Comment (junghans) [inline sci-libs/nipype/nipype-9999.ebuild]: These things could be prevented by running `repoman` (inside `sci-libs/nipype`).

**Comment by junghans** [comment]
Things to be done:
* `sci-biology/fsl` (#118) needs to be merged first
* `sci-biology/spm` (#107) needs to be merged first
* `ChangeLog` missing (can be generated by `repoman commit`)
* the use flags should be mapped on some build options.  Otherwise there will be auto-magic dependencies if the use flags is disabled, but the dependency is still installed. (See https://blog.flameeyes.eu/2010/07/ever-present-automagic-dependencies)

**Comment by jlec** [comment]
Start using repoman! Otherwise it is hard to handle you contributions.

**Comment by TheChymera** [comment]
@junghans how do I "map the flags on some build options"?

**Comment by marbre** [comment]
One possible solution would be the removal of the lines

> config.add_subpackage('fsl') 
> config.add_subpackage('spm')

from *nipype/interfaces/setup.py*. Anyway there might be much better solutions to pass this somehow to the main *setup.py*.

**Comment by TheChymera** [comment]
@marbre I don't get it - what do 
```
config.add_subpackage('fsl') 
config.add_subpackage('spm')
```
do that we do not want done?

**Comment by marbre** [comment]
@TheChymera 
*config.add_subpackage('fsl')* adds the subpackage fsl to *nipype*, resulting in support for fsl. So if the useflag fsl is deselected (-fsl), this subpackage shouldn't be added. Otherwise it results in auto-magic dependencies as @junghans noted. Same for spm.

**Comment by TheChymera** [comment]
1) Why is it a problem that all of nipype (including fsl/spm support) gets merged? Does this pull in other packages? I doubt that - so what exactly do you men by dependencies? Nipype is already in the overlay, no, nobdy brought it up when this got pulled :-/

2) I can't seem to run `repoman commit`:

```
chymera@zenbookhost ~/src/sci/sci-libs/nipype $ repoman commit

*** the local copy of metadata.dtd needs to be refetched, doing that now

Traceback (most recent call last):
  File "/usr/lib/python-exec/python2.7/repoman", line 1164, in <module>
    if not fetch_metadata_dtd():
  File "/usr/lib/python-exec/python2.7/repoman", line 1130, in fetch_metadata_dtd
    prefix='metadata.dtd.', dir=destdir)
  File "/usr/lib64/python2.7/tempfile.py", line 308, in mkstemp
    return _mkstemp_inner(dir, prefix, suffix, flags)
  File "/usr/lib64/python2.7/tempfile.py", line 239, in _mkstemp_inner
    fd = _os.open(file, flags, 0600)
OSError: [Errno 13] Permission denied: '/usr/portage/distfiles/metadata.dtd.S_B_4i'
```

**Comment by marbre** [comment]
1) Following your argumentation, you're pull request is useless as nipype builds with support for fsl and spm anyway. Thus use flags would be required. Anyway, it is the clean way to build features only if use flags are selected and only if they are selected.
Otherwise, this results in automagic dependencies and nipype will get into trouble as soon as fsl and/or spm are unmerged.
2) You're not in portage group.

**Comment by TheChymera** [comment]
1) My pull request would pull in two external packages - each required by nipype to actually do stuff. Some internal nipype interface functions are now also considered deps? How will nipype get into (more) trouble with these use flags than without them if fsl or spm get unmerged? I think messing with upstream internals is a bad idea.

2) Is it safe to add my user to the portage group? Can I just navigate with root to that dir and make root do this? Could that lead to permission issues?

**Comment by junghans** [comment]
0) I am not quite sure what the point of this pull request is, if neither `sci-biology/fsl` nor ` sci-biology/spm` are part of the sci overlay yet.
1) What we were getting at was, that if e.g. `sci-biology/fsl` was installed `sci-libs/nipype` would use it independently on whether the use `fsl` use flag is enabled not. That can lead to breakages (if e.g `sci-biology/fsl` gets uninstalled), because portage won't know for disabled `fsl` that `sci-libs/nipype` was actually depending on `biology/fsl`. Basically there are two solutions to that a) make the dependency unconditional or b) explicitly disable the use of the dependency (independently from it installation state) if the use flags is disable.
2) That is up to you, but I think it is safer to add a single user account (you trust!) to the portage group and running `repoman` with (lower) user permission than running all the command executed by `repoman` with root permissions. To avoid permission issues you should not mix the two schemes!

**Comment by TheChymera** [comment]
0) The point is to provide the prerequisites for pulling spm and fsl if you install nipype. In principle nipype is rather useless without spm, fsl (or afni).  

1) 
>if e.g. sci-biology/fsl was installed sci-libs/nipype would use it independently on whether the use fsl use flag is enabled not

**yes**. 

> That can lead to breakages (if e.g sci-biology/fsl gets uninstalled), because portage won't know for disabled fsl that sci-libs/nipype was actually depending on biology/fsl

**I don't understand** - please explain the second part of that phrase

**Comment by junghans** [comment]
0) That is clear, but where would users get `sci-biology/fsl` and `sci-libs/nipype` from? They aren't part of the science overlay or the standard gentoo tree yet!

1) Portage only know about dependencies from the listing in `RDEPEND` and `DEPEND`. So if the `fsl` use flag is disabled, `sci-biology/fsl` isn't a  dependency of `sci-libs/nipype`. And portage might depclean `sci-biology/fsl` without knowing that `sci-libs/nipype` is compiling against it and will break if sci-biology/fsl` get removed. Try the following example:
```
flaggie sci-libs/nipype -fsl
emerge sci-biology/fsl
emerge sci-libs/nipype
```
Now `sci-libs/nipype` is installed with `fsl` support, because it was found a compile time, but without portage knowing that. So if from some reason `sci-biology/fsl` get removed from the world file, depclean will remove it:
```
emerge --deselect sci-biology/fsl
emerge --depclean
```
Now you have a broken version of `sci-libs/nipype` without knowing it.

**Comment by TheChymera** [comment]
0) I have fsl in my overlay for a while now, so does @kiwifb , if anything I think it is preferable for the user to get a warning about a required package that does not exist, instead of nipype quietly installing without any backends to be used with.

1) I understand all that very well - **but** the same would happen without the flags existing. As it is now, nipype gets compiled with fsl support, without the user or portage even hearing about fsl.

**Comment by junghans** [comment]
0) We could add repository dependencies in `metadata/layout.conf`, but wouldn't it be much easier to merge these two packages?

1) True, but only if and only if, the user has installed `fsl` before manually. And that is pretty much the same as installing stuff outside portage (e.g. in `/usr/local`), which Gentoo allows, but doesn't officially support.

My take on this pull request so far is, get `sci-biology/fsl` (#118) and `sci-biology/spm` (#107) merged and we can go on. Otherwise, I don't see a point in pursuing this as it breaks more things than it fixes.

If you really really think the current version of `sci-libs/nipype` is useless without `sci-biology/fsl` and `sci-biology/spm`, create a `README.gentoo` using `readme.gentoo.eclass` to tell the user that there is a more versatile version of `sci-libs/nipype` in your overlay.

**Comment by TheChymera** [comment]
Roger that - in the mean time, let's get the rest evened out so that as soon as fsl and spm are done we can merge this.

Repoman won't commit:

```
>>> Creating Manifest for /home/chymera/src/sci/sci-libs/nipype
  IUSE.invalid                  2
   sci-libs/nipype/nipype-9999.ebuild: fsl
   sci-libs/nipype/nipype-9999.ebuild: spm
  dependency.badmasked          1
Please fix these important QA issues first.
RepoMan sez: "Make your QA payment on time and you'll never see the likes of me."
```

I take this to mean I have some errors in my ebuild? with my USE flags perhaps... but what exactly is wrong?

**Comment by marbre** [comment]
**man repoman**

>  IUSE.invalid
    This ebuild has a variable in IUSE that is not in the use.desc or its metadata.xml file

>  dependency.badmasked
Masked ebuilds with unsatisfied dependencies (matched against *all* ebuilds)


That is because the useflags are not in metadata.xml and because fsl and spm are not yet available (bad dependency).

**Comment by TheChymera** [comment]
where do I find `metadata.xml`?

**Comment by marbre** [comment]
Beside the ebuild.

**Comment by TheChymera** [comment]
ok, I added the USE flag metadata - but now repoman tells me:

```
zenbookhost nipype # repoman commit

RepoMan scours the neighborhood...
>>> Creating Manifest for /home/chymera/src/sci/sci-libs/nipype
  dependency.badmasked          1
   sci-libs/nipype/nipype-9999.ebuild: DEPEND: **(empty profile) ['sci-biology/fsl', 'sci-biology/spm']
RepoMan sez: "Doing nothing is not always good for QA."

(Didn't find any changed files...)
```

Is this good? It sounds like repoman took no notice of my change, but then why does it not give me the same error message as before?

**Comment by junghans** [comment]
Well that basically means `sci-biology/fsl` and  `sci-biology/spm` could not be found.

**Comment by TheChymera** [comment]
@junghans ok, so now this is good to go pending inclusion of FSL an SPM?

**Comment by junghans** [comment]
Yes, except there is still a `ChangeLog` entry missing.

**Comment by TheChymera** [comment]
How do I get that changelog entry?

**Comment by junghans** [comment]
Use `repoman -m "Some message" commit` ;-)

**Comment by TheChymera** [comment]
@junghans ok, I did that, now how do I push it to here? after committing via repoman git won't push or commit anything, because it sees no changes.

**Comment by junghans** [comment]
Ok, if you already committed the change, you have to do `git reset HEAD^` before.

**Comment by TheChymera** [comment]
Ok, so apparently repoman won't run as root because I am root:
```
zenbookhost nipype # repoman -m "added changeog" commit 
!!! Section 'funtoo' in repos.conf has name different from repository name 'funtoo-overlay' set inside repository

RepoMan scours the neighborhood...
>>> Creating Manifest for /home/chymera/src/sci/sci-libs/nipype
  dependency.badmasked          1
   sci-libs/nipype/nipype-9999.ebuild: DEPEND: **(empty profile) ['sci-biology/fsl', 'sci-biology/spm']
[CRITICAL] Please set ECHANGELOG_USER or run as non-root
!!! Updating the ChangeLog failed
```
and it won't run as user because I do not have a key
```
chymera@zenbookhost ~/src/sci/sci-libs/nipype $ repoman -m "added changelog" commit
!!! Section 'funtoo' in repos.conf has name different from repository name 'funtoo-overlay' set inside repository

RepoMan scours the neighborhood...
>>> Creating Manifest for /home/chymera/src/sci/sci-libs/nipype
  dependency.badmasked          1
   sci-libs/nipype/nipype-9999.ebuild: DEPEND: **(empty profile) ['sci-biology/fsl', 'sci-biology/spm']

--- ./ChangeLog
+++ ./ChangeLog
@@ -2 +2 @@
-# Copyright 1999-2014 Gentoo Foundation; Distributed under the GPL v2
+# Copyright 1999-2015 Gentoo Foundation; Distributed under the GPL v2
@@ -4,0 +5,3 @@
+  11 Feb 2015;  <chymera@gentoo.org> metadata.xml:
+  added changelog
+

* 2 files being committed... 
>>> Creating Manifest for /home/chymera/src/sci/sci-libs/nipype
gpg: skipped "Horea Christian <h.chr@mail.ru>": No secret key
gpg: signing failed: No secret key
error: gpg failed to sign the data
fatal: failed to write commit object
!!! Exiting on git (shell) error code: 128
```
what? seriously, is repoman really worth this trouble?

**Comment by junghans** [comment]
You can add `-sign` to your `FEATURES` in `make.conf`.

**Comment by TheChymera** [comment]
is that advisable? what do I get from signing stuff?

**Comment by junghans** [comment]
You get nothing from signing, but all the other users can check that the commit was actually done by you (or somebody who had you secret key). Anyhow, we have `sign-commits = true` in `metadata/layout.conf`, which I think overwrites `FEATURES=-sign`, so you will need a gpg setup no matter what. See https://wiki.gentoo.org/wiki/GnuPG for details to do that.

**Comment by TheChymera** [comment]
Was this always the case? I have gotten quite a few pull requests accepted here, but I never had to mess around with keys before.

**Comment by junghans** [comment]
As added back in Aug. 2012:
```
commit 0ba92ff936d85c435917035deeb4b2f98bcef180
Author: Justin Lecher <jlec@gentoo.org>
Date:   Thu Aug 30 10:43:52 2012 +0200

    enabled commit signing

diff --git a/metadata/layout.conf b/metadata/layout.conf
index 46ccb35..081a326 100644
--- a/metadata/layout.conf
+++ b/metadata/layout.conf
@@ -14,7 +14,7 @@ thin-manifests = true
 sign-manifests = false

 # sign commits using git mechanisms
-sign-commits = false
+sign-commits = true

 # Make egencache generate newer (more reliable)
 # md5-dict cache format (see bug #409445).
```

**Comment by TheChymera** [comment]
It still won't sign it, though I plainly have a key now: 

```
chymera@zenbookhost ~/src/neurogentoo/sci-libs/nipype $ repoman -m "added changelog" commit
!!! Section 'neurogentoo' in repos.conf has name different from repository name 'science' set inside repository

RepoMan scours the neighborhood...
>>> Creating Manifest for /home/chymera/src/neurogentoo/sci-libs/nipype
  dependency.badmasked          1
   sci-libs/nipype/nipype-9999.ebuild: DEPEND: **(empty profile) ['sci-biology/fsl', 'sci-biology/spm']
* 3 files being committed... 
gpg: skipped "Horea Christian <h.chr@mail.ru>": No secret key
gpg: signing failed: No secret key
error: gpg failed to sign the data
fatal: failed to write commit object
!!! Exiting on git (shell) error code: 128
chymera@zenbookhost ~/src/neurogentoo/sci-libs/nipype $ gpg --list-keys
/home/chymera/.gnupg/pubring.gpg
--------------------------------
pub   2048R/C7DE3E5F 2015-02-13
uid       [ultimate] Horea Christian (Chymera) <h.chr@mail.ru>
uid       [ultimate] Horea Christian (Accept no substitutes) <horea.christ@gmail.com>
sub   2048R/D30FEC8D 2015-02-13
```

Again, is all of this really necessary? How about we make pushing to here as easy as possible so that one can invest what little time one finds for contributing into good ebuilds rather than overhead and superfluous security features?

**Comment by jlec** [comment]
check gpg -K whether you really have the private key. Is your user.signingkey setting in git set to the correct key? See this for more information http://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work

And please stop complaining the way you are right. Simply the fact that you are developing as root shows that there is still some things to learn for you. Secondly, you never used repoman, as your just now facing those problems. Repoman isn't a recommendation but a must.

**Comment by TheChymera** [comment]
Where did you get the notion that I am developing as root? The only think which would make me resort to root atm is `repoman`. 

It seems I have the key, and I also told git to enable it. But now repoman doesn't want to do anything anymore?

```
chymera@zenbookhost ~/src/neurogentoo/dev-tex/pythontex $ gpg -K
/home/chymera/.gnupg/secring.gpg
--------------------------------
sec   2048R/C7DE3E5F 2015-02-13
uid                  Horea Christian (Accept no substitutes) <horea.christ@gmail.com>
uid                  Horea Christian (Chymera) <h.chr@mail.ru>
ssb   2048R/D30FEC8D 2015-02-13

chymera@zenbookhost ~/src/neurogentoo/dev-tex/pythontex $ gpg --list-keys
/home/chymera/.gnupg/pubring.gpg
--------------------------------
pub   2048R/C7DE3E5F 2015-02-13
uid       [ultimate] Horea Christian (Chymera) <h.chr@mail.ru>
uid       [ultimate] Horea Christian (Accept no substitutes) <horea.christ@gmail.com>
sub   2048R/D30FEC8D 2015-02-13

chymera@zenbookhost ~/src/neurogentoo/dev-tex/pythontex $ git config --global user.signingkey C7DE3E5F
chymera@zenbookhost ~/src/neurogentoo/dev-tex/pythontex $ cd ../..
chymera@zenbookhost ~/src/neurogentoo $ git checkout patch-1
Switched to branch 'patch-1'
Your branch is behind 'origin/patch-1' by 3 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)
chymera@zenbookhost ~/src/neurogentoo $ cd sci-libs/nipype/
chymera@zenbookhost ~/src/neurogentoo/sci-libs/nipype $ repoman -m "added changelog" commit
!!! Repository 'science' has sync-type attribute set to unsupported value: 'git'
!!! Installed sync-types are: '['rsync', 'websync']'
!!! Section 'neurogentoo' in repos.conf has name different from repository name 'science' set inside repository
!!! Repository 'science' has sync-type attribute set to unsupported value: 'git'
!!! Installed sync-types are: '['rsync', 'websync']'

RepoMan scours the neighborhood...
>>> Creating Manifest for /home/chymera/src/neurogentoo/sci-libs/nipype
RepoMan sez: "Doing nothing is not always good for QA."

(Didn't find any changed files...)

chymera@zenbookhost ~/src/neurogentoo/sci-libs/nipype $ git reset HEAD^
Unstaged changes after reset:
M	sci-libs/acml/ChangeLog
M	sci-libs/acml/Manifest
chymera@zenbookhost ~/src/neurogentoo/sci-libs/nipype $ repoman -m "added changelog" commit
!!! Repository 'science' has sync-type attribute set to unsupported value: 'git'
!!! Installed sync-types are: '['rsync', 'websync']'
!!! Section 'neurogentoo' in repos.conf has name different from repository name 'science' set inside repository
!!! Repository 'science' has sync-type attribute set to unsupported value: 'git'
!!! Installed sync-types are: '['rsync', 'websync']'

RepoMan scours the neighborhood...
>>> Creating Manifest for /home/chymera/src/neurogentoo/sci-libs/nipype
RepoMan sez: "Doing nothing is not always good for QA."

(Didn't find any changed files...)

chymera@zenbookhost ~/src/neurogentoo/sci-libs/nipype $ 
```

**Comment by junghans** [comment]
Try setting PORTAGE_GPG_KEY="C7DE3E5F" in your environment or add it to `make.conf`.

**Comment by TheChymera** [comment]
@junghans 
```
zenbookhost nipype # export PORTAGE_GPG_KEY="C7DE3E5F"
zenbookhost nipype # repoman commit 
!!! Section 'neurogentoo' in repos.conf has name different from repository name 'science' set inside repository

RepoMan scours the neighborhood...
>>> Creating Manifest for /home/chymera/src/neurogentoo/sci-libs/nipype
!!! The following files are in your local tree but are not added to the master
!!! tree. Please remove them from the local tree or add them to the master tree.
    ./]
```

doesn't seem to have worked either...

**Comment by junghans** [comment]
Now the gpg error message is gone, but repoman is complaining that there is an unknown file named `]` in the path. Check that with `git status`!

**Comment by TheChymera** [comment]
@junghans I am on another branch (which I recently rebased) and I can't go back, to work on this because:

```
chymera@zenbookhost ~/src/neurogentoo $ git checkout patch-1
error: The following untracked working tree files would be overwritten by checkout:
	sci-biology/bowtie/files/bowtie-2.1.0-buildsystem.patch
	sci-biology/embassy-cbstools/files/embassy-cbstools-1.0.0-r2.patch
	sci-biology/embassy-domainatrix/files/embassy-domainatrix-0.1.0-r5.patch
	sci-biology/embassy-domalign/files/embassy-domalign-0.1.0-r5.patch
	sci-biology/embassy-domsearch/files/embassy-domsearch-0.1.0-r5.patch
	sci-biology/embassy-emnu/files/embassy-emnu-1.05-r7.patch
	sci-biology/embassy-esim4/files/embassy-esim4-1.0.0-r7.patch
	sci-biology/embassy-hmmer/files/embassy-hmmer-2.3.2-r4.patch
	sci-biology/embassy-iprscan/files/embassy-iprscan-4.3.1-r2.patch
	sci-biology/embassy-memenew/ChangeLog
	sci-biology/embassy-memenew/files/embassy-memenew-4.0.0-r1.patch
	sci-biology/embassy-memenew/metadata.xml
	sci-biology/embassy-mira/ChangeLog
	sci-biology/embassy-mira/files/embassy-mira-2.8.2-r2.patch
	sci-biology/embassy-mira/metadata.xml
	sci-biology/embassy-mse/files/embassy-mse-3.0.0.patch
	sci-biology/embassy-phylipnew/files/embassy-phylipnew-3.69.patch
	sci-biology/embassy-signature/files/embassy-signature-0.1.0-r5.patch
	sci-biology/embassy-structure/files/embassy-structure-0.1.0-r5.patch
	sci-biology/embassy-topo/files/embassy-topo-2.0.0.patch
	sci-biology/embassy-vienna/files/embassy-vienna-1.7.2-r2.patch
	sci-biology/emboss/files/ax_lib_mysql.m4
	sci-biology/emboss/files/emboss-6.3.1_p4-r2_Makefile.patch
	sci-biology/emboss/files/emboss-6.3.1_p4-r2_compilations-paths.patch
	sci-biology/emboss/files/emboss-6.3.1_p4-r2_libtool.patch
	sci-biology/emboss/files/emboss-6.3.1_p4-r2_plcol.patch
	sci-biology/emboss/files/emboss-6.3.1_p4-r2_rename_plwid_to_plwidth.patch
	sci-biology/emboss/files/emboss-6.6.0_plcol.patch
	sci-biology/emboss/files/emboss-README.Gentoo-2
Please move or remove them before you can switch branches.
Aborting
```

What can I do? How could this have happened?

**Comment by junghans** [comment]
I am not 100% sure how you arrived at that state, but basically you will have to remove those files otherwise a checkout isn't possible. You can use `git clean` to purge them.

**Comment by TheChymera** [comment]
ok, does this mean all is ok except that spm and fsl still need to be pulled?

```
chymera@zenbookhost ~/src/neurogentoo/sci-libs/nipype $ repoman commit
!!! Section 'neurogentoo' in repos.conf has name different from repository name 'science' set inside repository

RepoMan scours the neighborhood...
>>> Creating Manifest for /home/chymera/src/neurogentoo/sci-libs/nipype
  dependency.badmasked          1
   sci-libs/nipype/nipype-9999.ebuild: DEPEND: **(empty profile) ['sci-biology/fsl', 'sci-biology/spm']
RepoMan sez: "Doing nothing is not always good for QA."

(Didn't find any changed files...)
```

**Comment by junghans** [comment]
Yes, it means to make repoman happy `sci-biology/fsl` and `sci-biology/spm` need to be merged first.
Second, to generate the missing changelog, you will have to redo that commit:
```
git reset HEAD^
git commit -m "added fsl and spm use flags" commit
```

**Comment by TheChymera** [comment]
@junghans are you sure this is right? `git commit -m "added fsl and spm use flags" commit` - with a second `commit` at the end? Also I remember someone (maybe even you?) recently telling me that I should only commit with repoman. Isn't repoman what is making the `ChangeLog`?

**Comment by junghans** [comment]
You are absolutely right, I mistype, I was supposed to be:
```
git reset HEAD^
repoman -m "added fsl and spm use flags" commit
```
But hey, it seems you finally got the hang on repoman :-)

**Comment by TheChymera** [comment]
I guess you will also want the `ChangeLog` to contain my name and email address?

```
chymera@zenbookhost ~/src/neurogentoo/sci-libs/nipype $ git reset HEAD^
chymera@zenbookhost ~/src/neurogentoo/sci-libs/nipype $ repoman -m "added fsl and spm use flags" commit
!!! Section 'neurogentoo' in repos.conf has name different from repository name 'science' set inside repository

RepoMan scours the neighborhood...
>>> Creating Manifest for /home/chymera/src/neurogentoo/sci-libs/nipype
  dependency.badmasked          1
   sci-libs/nipype/nipype-9999.ebuild: DEPEND: **(empty profile) ['sci-biology/fsl', 'sci-biology/spm']
RepoMan sez: "Doing nothing is not always good for QA."

(Didn't find any changed files...)

chymera@zenbookhost ~/src/neurogentoo/sci-libs/nipype $ git push origin
To https://github.com/TheChymera/neurogentoo.git
 ! [rejected]        patch-1 -> patch-1 (non-fast-forward)
error: failed to push some refs to 'https://github.com/TheChymera/neurogentoo.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
chymera@zenbookhost ~/src/neurogentoo/sci-libs/nipype $ git pull origin
Updating 42d4639..bbc0ef5
Fast-forward
chymera@zenbookhost ~/src/neurogentoo/sci-libs/nipype $ git push origin
Everything up-to-date
chymera@zenbookhost ~/src/neurogentoo/sci-libs/nipype $ repoman -m "added fsl and spm use flags" commit
!!! Section 'neurogentoo' in repos.conf has name different from repository name 'science' set inside repository

RepoMan scours the neighborhood...
>>> Creating Manifest for /home/chymera/src/neurogentoo/sci-libs/nipype
  dependency.badmasked          1
   sci-libs/nipype/nipype-9999.ebuild: DEPEND: **(empty profile) ['sci-biology/fsl', 'sci-biology/spm']
RepoMan sez: "Doing nothing is not always good for QA."

(Didn't find any changed files...)

chymera@zenbookhost ~/src/neurogentoo/sci-libs/nipype $ git pull origin
Already up-to-date.
chymera@zenbookhost ~/src/neurogentoo/sci-libs/nipype $ git push origin
Everything up-to-date
chymera@zenbookhost ~/src/neurogentoo/sci-libs/nipype $ echo $ECHANGELOG_USER
Horea Christian <h.chr@mail.ru>
chymera@zenbookhost ~/src/neurogentoo/sci-libs/nipype $ cat ChangeLog 
# ChangeLog for sci-libs/nipype
# Copyright 1999-2015 Gentoo Foundation; Distributed under the GPL v2
# $Header: $

  20 Feb 2015;  <chymera@gentoo.org> metadata.xml:
  added fsl and spm use flags

  06 Jan 2014; Justin Lecher <jlec@gentoo.org> nipype-9999.ebuild,
  metadata.xml:
  Switch from git-2 to git-r3
``` 

what now?

**Comment by junghans** [comment]
After `git reset HEAD^` you need to remove the wrong Changelog entry and re-commit using repoman.

Anyhow, because the change is so small it might be easier to reset to master
```
git checkout patch-1
git reset --hard origin/master
```
and re-apply the diff
```
cd sci-libs/nipype
wget -O - https://github.com/gentoo-science/sci/pull/361.diff | git apply -
```
but trash the `ChangLog` by
```
git checkout -- ChangeLog
```
and re-commit using:
```
repoman -m "added fsl and spm use flags" commit
```
and then force-push the change to github
```
git push -f origin patch-1:patch-1
```

**Pull request closed without merging**